pinot-azure-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java

```diff
@@ -116,13 +117,24 @@ public long length(URI fileUri) throws IOException {
 
   @Override
   public String[] listFiles(URI fileUri) throws IOException {
-    // list directory contents
-    List<DirectoryEntry> list = _adlStoreClient.enumerateDirectory(fileUri.getPath(), Integer.MAX_VALUE);
-    List<String> fileNames = new ArrayList<>();
-    for (DirectoryEntry directoryEntry : list) {
-      fileNames.add(directoryEntry.name);
+    DirectoryEntry rootDir = _adlStoreClient.getDirectoryEntry(fileUri.getPath());
+    if (rootDir == null) {
+      return new String[0];
```

Comment:
Save the empty string in a static final, so we don't new a garbage with each call.